src/binary_agreement/mod.rs

```diff
@@ -148,11 +149,11 @@ pub struct Message {
 }
 
 // NOTE: Extending rand_derive to correctly generate random values from boxes would make this
-// implementation obsolete; however at the time of this writing, `rand::Rand` is already deprecated
+// implementation obsolete; however at the time of this writing, `rand_derive` is already deprecated
 // with no replacement in sight.
-impl rand::Rand for MessageContent {
-    fn rand<R: rand::Rng>(rng: &mut R) -> Self {
-        let message_type = *rng.choose(&["sbvb", "conf", "term", "coin"]).unwrap();
+impl Distribution<MessageContent> for Standard {
```

Comment:
These are slated for removal anyway, as the previous `rand::Rand` implementation was only used for random adversary testing. We could port the little bit of code that uses this (if it's still present) and remove these altogether. Maybe this is for a separate PR though.

Reply:
I'm not sure what you mean: Don't we want to keep the random adversaries?

I guess it would be cleaner to make the random adversary put together the messages by hand, so these `impl`s don't pollute the production code. In a separate PR, though, yes.

Reply:
Yes, we do want the random adversaries, but we would use `proptest`'s facilities for this in the future, generating these messages outside the core library.